code/datums/elements/tool_bump.dm

```diff
@@ -0,0 +1,87 @@
+/**
+ * Tool bump bespoke element
+ *
+ * Attacks attached atom with the bumper's held object only if it has a specific tool_behaviour or is a specific type of item (e.g. fireaxe).
+ */
+/datum/element/tool_bump
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	///Behaviours to look for in bumper's held objects before attacking the attached atom with one.
+	var/list/tool_behaviours
+	///Can we use this tool to perform this behaviour with our off-hand (e.g. mining)?
+	var/require_active_hand = TRUE
+	///Tool types to look for in bumper's held objects before attacking the attached atom with one.
+	var/list/tool_items
+
+/datum/element/tool_bump/Attach(datum/target, tool_behaviours = null, require_active_hand = TRUE, tool_items = null)
+	. = ..()
+
+	if(!isatom(target) || isarea(target))
+		return ELEMENT_INCOMPATIBLE
+	LAZYADD(src.tool_behaviours, tool_behaviours)
+	src.require_active_hand = require_active_hand
+	LAZYADD(src.tool_items, tool_items)
```

Comment:
Why make these lists lazy? couldn't you just set them to be equal to the args?

Reply:
I assumed that having a list on every mining turf would be bad for performance and that making it a lazy list would be better. Is it different for elements? I'm not sure what you mean by set them to be equal to the args. Just `src.tool_items += tool_items`?

Reply:
A bespoke element like the one you've created is different from a component. 
For each set of arguments, a unique element is created.
So if you add an element twice with the same args, no duplication happens.

Regardless of that. tool_behaviors and tool_items default to null, so why not just set src.tool_behaviors and src.tool_items directly? All a non filled lazy list is is a null value anyhow

Reply:
Okay, done